### ISSUE TITLE
Traefik's server_cert_sans_dns has unexpected contents

### ISSUE THREAD

**Issue by PietroPasotti** [canonical/traefik-k8s-operator#347]
### Bug Description

TLDR: "It works on my virtual machine"

Traefik issues a CSR for "`Creating CSR for darkstar.kpn with DNS ['darkstar.kpn'] and IPs [] `" because `Traefik.server_cert_sans_dns` is returning `['darkstar.kpn']` i.e. 

Supposing the app IP is  "10.64.140.43"
```python
import socket
socket.gethostbyaddr("10.64.140.43")   
('darkstar.kpn', [], ['10.64.140.43'])     
```

Note that `darkstar` is the name of my machine.

In a VM, you get

```python
>>> socket.gethostbyaddr("10.233.117.142")
('10-233-117-142.kubernetes.default.svc.cluster.local', [], ['10.233.117.142'])
```

Traefik doesn't expect 'darkstar.kpn' and therefore we have a mismatch between the SANs in the certs issued to traefik by self-signed-certificates.

Note: in another case, (the VM's network was bridged to the host with e.g. `--network enp1s0`) we got:
```python
>>> socket.gethostbyaddr("192.168.1.250")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
socket.herror: [Errno 1] Unknown host
```

![image](https://github.com/canonical/traefik-k8s-operator/assets/6230162/43e7bb5b-be7a-4314-8c55-f81f513be946)

### To Reproduce

n/a

### Environment

n/a

### Relevant log output

```shell
n/a
```

**Comment by mmkay** [comment]
We could try setting traefik's external hostname to `darkstar.kpn` to see if that helps. If this is the issue, we could try to document it or detect this scenario and change the logic around it.
